source/apiVolontaria/volunteer/views.py

```diff
@@ -344,10 +345,36 @@ def get_queryset(self):
             queryset = queryset.filter(user__username=username)
         return queryset
 
+    def get_serializer_class(self):
+        # If authenticated user is admin
+        if self.request.user.is_superuser:
+            return serializers.ParticipationAdminSerializer
+        else:
+            return serializers.ParticipationBasicSerializer
+
     # A user can only create participations for himself
     # This auto-fills the 'user' field of the Participation object.
     def perform_create(self, serializer):
-        serializer.save(user=self.request.user)
+        user_id = self.request.data.get('user', None)
+        user_obj = self.request.user
+
+        if user_id:
+            if self.request.user.is_superuser:
+                # We have the permission to override the user
+                user_obj = User.objects.filter(id=user_id).first()
+
+        serializer.save(user=user_obj)
+
+    def post(self, request, *args, **kwargs):
+        user_id = self.request.data.get('user', None)
+
+        if user_id and not self.request.user.is_superuser:
```

Comment:
If you do this kind of hardcoded verification, manager will not be able to create participation with custom user.